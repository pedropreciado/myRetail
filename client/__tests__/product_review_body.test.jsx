import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import StarRating from 'react-star-ratings';

import ProductReviewBody from '../components/product/show/product_review_body';

configure({ adapter: new Adapter() });

const review = {
  datePosted: 'Mon Mar 11 13: 13: 55 UTC 2013',
  overallRating: '4',
  review: 'plop',
  screenName: 'me',
  title: 'a good review'
};

describe('ProductReviewBody', () => {
  let productReviewBody = shallow(
    <ProductReviewBody review={review}/>
  );
  
  it('renders correctly', () => {
    expect(productReviewBody).toMatchSnapshot();
  });
  
  it('shows ratings as stars', () => {
    let header = productReviewBody.find('.pro-con-body-header');

    expect(header.props().children[0].type).toBe(StarRating);
  });
  
  it('shows title', () => {
    let header = productReviewBody.find('.pro-con-body-header');
    
    expect(header.props().children[1].props.children).toBe('a good review');
  });
  
  it('shows review', () => {
    let reviewContainer = productReviewBody.find('.pro-con-review').at(0);
    
    expect(reviewContainer.props().children).toBe('plop');
  });
});