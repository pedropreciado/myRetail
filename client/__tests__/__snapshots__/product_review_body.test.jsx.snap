// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductReviewBody renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReviewBody
    review={
        Object {
            "datePosted": "Mon Mar 11 13: 13: 55 UTC 2013",
            "overallRating": "4",
            "review": "plop",
            "screenName": "me",
            "title": "a good review",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="pro-con-body other-review"
>
        <div
                className="pro-con-body-header"
        >
                <StarRatings
                        changeRating={null}
                        gradientPathName=""
                        ignoreInlineStyles={false}
                        name="rating"
                        numberOfStars={5}
                        rating={4}
                        starDimension="10px"
                        starEmptyColor="rgb(203, 211, 227)"
                        starHoverColor="rgb(230, 67, 47)"
                        starRatedColor="red"
                        starSpacing="2.5px"
                        svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                        svgIconViewBox="0 0 51 48"
                        typeOfWidget="Star"
                />
                <a>
                        a good review
                </a>
        </div>
        <p
                className="pro-con-review"
        >
                plop
        </p>
        <a
                className="pro-con-review author"
        >
                me Mar 11, 55
        </a>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="pro-con-body-header"
>
            <StarRatings
                        changeRating={null}
                        gradientPathName=""
                        ignoreInlineStyles={false}
                        name="rating"
                        numberOfStars={5}
                        rating={4}
                        starDimension="10px"
                        starEmptyColor="rgb(203, 211, 227)"
                        starHoverColor="rgb(230, 67, 47)"
                        starRatedColor="red"
                        starSpacing="2.5px"
                        svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                        svgIconViewBox="0 0 51 48"
                        typeOfWidget="Star"
            />
            <a>
                        a good review
            </a>
</div>,
          <p
            className="pro-con-review"
>
            plop
</p>,
          <a
            className="pro-con-review author"
>
            me Mar 11, 55
</a>,
        ],
        "className": "pro-con-body other-review",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <StarRatings
                changeRating={null}
                gradientPathName=""
                ignoreInlineStyles={false}
                name="rating"
                numberOfStars={5}
                rating={4}
                starDimension="10px"
                starEmptyColor="rgb(203, 211, 227)"
                starHoverColor="rgb(230, 67, 47)"
                starRatedColor="red"
                starSpacing="2.5px"
                svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                svgIconViewBox="0 0 51 48"
                typeOfWidget="Star"
/>,
              <a>
                a good review
</a>,
            ],
            "className": "pro-con-body-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "changeRating": null,
                "gradientPathName": "",
                "ignoreInlineStyles": false,
                "name": "rating",
                "numberOfStars": 5,
                "rating": 4,
                "starDimension": "10px",
                "starEmptyColor": "rgb(203, 211, 227)",
                "starHoverColor": "rgb(230, 67, 47)",
                "starRatedColor": "red",
                "starSpacing": "2.5px",
                "svgIconPath": "m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z",
                "svgIconViewBox": "0 0 51 48",
                "typeOfWidget": "Star",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "a good review",
              },
              "ref": null,
              "rendered": "a good review",
              "type": "a",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "plop",
            "className": "pro-con-review",
          },
          "ref": null,
          "rendered": "plop",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "me Mar 11, 55",
            "className": "pro-con-review author",
          },
          "ref": null,
          "rendered": "me Mar 11, 55",
          "type": "a",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="pro-con-body other-review"
>
          <div
                    className="pro-con-body-header"
          >
                    <StarRatings
                              changeRating={null}
                              gradientPathName=""
                              ignoreInlineStyles={false}
                              name="rating"
                              numberOfStars={5}
                              rating={4}
                              starDimension="10px"
                              starEmptyColor="rgb(203, 211, 227)"
                              starHoverColor="rgb(230, 67, 47)"
                              starRatedColor="red"
                              starSpacing="2.5px"
                              svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                              svgIconViewBox="0 0 51 48"
                              typeOfWidget="Star"
                    />
                    <a>
                              a good review
                    </a>
          </div>
          <p
                    className="pro-con-review"
          >
                    plop
          </p>
          <a
                    className="pro-con-review author"
          >
                    me Mar 11, 55
          </a>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="pro-con-body-header"
>
              <StarRatings
                            changeRating={null}
                            gradientPathName=""
                            ignoreInlineStyles={false}
                            name="rating"
                            numberOfStars={5}
                            rating={4}
                            starDimension="10px"
                            starEmptyColor="rgb(203, 211, 227)"
                            starHoverColor="rgb(230, 67, 47)"
                            starRatedColor="red"
                            starSpacing="2.5px"
                            svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                            svgIconViewBox="0 0 51 48"
                            typeOfWidget="Star"
              />
              <a>
                            a good review
              </a>
</div>,
            <p
              className="pro-con-review"
>
              plop
</p>,
            <a
              className="pro-con-review author"
>
              me Mar 11, 55
</a>,
          ],
          "className": "pro-con-body other-review",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <StarRatings
                  changeRating={null}
                  gradientPathName=""
                  ignoreInlineStyles={false}
                  name="rating"
                  numberOfStars={5}
                  rating={4}
                  starDimension="10px"
                  starEmptyColor="rgb(203, 211, 227)"
                  starHoverColor="rgb(230, 67, 47)"
                  starRatedColor="red"
                  starSpacing="2.5px"
                  svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
                  svgIconViewBox="0 0 51 48"
                  typeOfWidget="Star"
/>,
                <a>
                  a good review
</a>,
              ],
              "className": "pro-con-body-header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "changeRating": null,
                  "gradientPathName": "",
                  "ignoreInlineStyles": false,
                  "name": "rating",
                  "numberOfStars": 5,
                  "rating": 4,
                  "starDimension": "10px",
                  "starEmptyColor": "rgb(203, 211, 227)",
                  "starHoverColor": "rgb(230, 67, 47)",
                  "starRatedColor": "red",
                  "starSpacing": "2.5px",
                  "svgIconPath": "m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z",
                  "svgIconViewBox": "0 0 51 48",
                  "typeOfWidget": "Star",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "a good review",
                },
                "ref": null,
                "rendered": "a good review",
                "type": "a",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "plop",
              "className": "pro-con-review",
            },
            "ref": null,
            "rendered": "plop",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "me Mar 11, 55",
              "className": "pro-con-review author",
            },
            "ref": null,
            "rendered": "me Mar 11, 55",
            "type": "a",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
